src/lib/input/input.ts

```diff
@@ -351,7 +355,18 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
    * Implemented as part of MatFormFieldControl.
    * @docs-private
    */
-  get shouldLabelFloat(): boolean { return this.focused || !this.empty; }
+  get shouldLabelFloat(): boolean {
+    if (this._isNativeSelect) {
+      // For multi select, float mat input label
+      // If single select has value or has a option label or html, float mat input label to avoid
+      // mat input label to overlap with the select content.
```

Comment:
Line-break and capitalization is off. Also, should this say "For native select" instead of "For multi select"?